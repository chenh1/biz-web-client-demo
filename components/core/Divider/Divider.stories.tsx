import type { DividerType } from './Divider';
import React from 'react';
import { Divider } from './Divider';
import { Box } from '../../containers/Box/Box';
import { Heading } from '../Heading/Heading';

type Args = {
  args: DividerType
}

const Template = (args: DividerType) =>  (
  <Box className={`auto-cols-auto auto-rows-auto ${args.orientation === "horizontal" ? 'grid-rows-5' : 'grid-cols-5'}`}>
    <Heading type="h3">Heading text</Heading>
    <Divider {...args}>Divider</Divider>
    <Heading type="h3">Heading text</Heading>
    <Divider {...args}>Divider</Divider>
    <Heading type="h3">Heading text</Heading>
  </Box>
)

export default {
  component: Template
}

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const HorizontalDivider: Args = {
  args: {
    orientation: "horizontal"
  },
};

export const VerticalDivider: Args = {
  args: {
    orientation: "vertical"
  },
};