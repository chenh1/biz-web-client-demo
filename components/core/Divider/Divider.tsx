import type { Orientation } from '../../constants/types';
import React, { useMemo } from 'react';

export type DividerType = { orientation?: Orientation, className?: string }

const getOrientationClass = (orientation: Orientation): string => {
  if (orientation === "horizontal") {
    return "w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white"
  } else {
    return "h-full w-0 border-0 border-l-[1px] border-night-black-300 dark:border-max-white"
  }
}

export const Divider: React.FC<DividerType> = ({ orientation = "horizontal", className = "" }) => {
  const computedClasses = useMemo(() => {
    return getOrientationClass(orientation);
  }, [orientation]);

  return (
    <hr className={`${computedClasses} ${className}`}/>
  )
};