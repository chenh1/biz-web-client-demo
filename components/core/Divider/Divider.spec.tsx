import { Divider } from "./Divider";
import { render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders a horizontal divider', async () => {
  const { asFragment } = render(
    <Divider orientation="horizontal"/>
  );

  expect(asFragment()).toMatchSnapshot();
});

it('renders a vertical divider', async () => {
  const { asFragment } = render(
    <Divider />
  );

  expect(asFragment()).toMatchSnapshot();
});