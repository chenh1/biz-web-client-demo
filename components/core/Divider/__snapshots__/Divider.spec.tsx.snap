// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a horizontal divider 1`] = `
<DocumentFragment>
  <hr
    class="w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white "
  />
</DocumentFragment>
`;

exports[`renders a vertical divider 1`] = `
<DocumentFragment>
  <hr
    class="w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white "
  />
</DocumentFragment>
`;
